bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/PercentageValue.java

```diff
@@ -131,20 +134,27 @@ public void update(Command command) throws IllegalArgumentException {
     }
 
     @Override
-    public String getMQTTpublishValue() {
+    public String getMQTTpublishValue(@Nullable String pattern) {
         if (state == UnDefType.UNDEF) {
             return "";
         }
-        // Formular: From percentage to custom min/max: value*span/100+min
+        // Formula: From percentage to custom min/max: value*span/100+min
         // Calculation need to happen with big decimals to either return a straight integer or a decimal depending on
         // the value.
-        return ((PercentType) state).toBigDecimal().multiply(BigDecimal.valueOf(span)).divide(BigDecimal.valueOf(100))
-                .add(BigDecimal.valueOf(min)).toString();
+        BigDecimal value = ((PercentType) state).toBigDecimal().multiply(span).divide(DB100, MathContext.DECIMAL128)
+                .add(min).stripTrailingZeros();
+
+        String formatPattern = pattern;
+        if (formatPattern == null) {
+            formatPattern = "%s";
+        }
+
+        return new DecimalType(value).format(formatPattern);
```

Comment:
This is the part, which is changed against the original PR.

Use of the `DecimalType.format` to do the real work.